.github/workflows/test.yaml

```diff
@@ -0,0 +1,60 @@
+# -----------------------------------------------------------------------------
+#   - invoked on push, pull_request, manual trigger, or schedule
+# -----------------------------------------------------------------------------
+name: test
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+  schedule: 
+    - cron: "0 8 * * 1"
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Print environment values
+        run: |
+          python --version
+          cat $GITHUB_ENV
+      - name: Update pip
+        run: |
+          python -m pip install --upgrade pip
+      - name: Install dependencies from lock file
+        run: |
+          pip install -r pip.lock.txt
+          pip install .
+      - name: Run pipeline
+        run: |
+          snakemake --forceall
+      - name: Install lint dependencies
+        run: |
+          pip install .[dev]
+      - name: Lint
```

Comment:
I'm guessing you're doing the linting after the run itself to test the generated notebooks?

Reply:
Not really -- I was thinking I didn't want to install the dev dependencies before doing the run. I should likely just make this two separate jobs.